### ISSUE TITLE
v3/processes/:guid/stats shows Desired memory, disk and fds quotas, not Actual state

### ISSUE THREAD

**Issue by sethboyles** [cloudfoundry/cloud_controller_ng#2098]
The response to v3 process stats endpoint returns information about memory and disk quota:

```
 {
      "type": "web",
      "index": 0,
      "state": "RUNNING",
      "usage": {
        "time": "2021-02-01T20:01:22+00:00",
        "cpu": 0.005951195409915775,
        "mem": 41677255,
        "disk": 136368128
      },
      "host": "10.244.0.138",
      "uptime": 2730,
      "mem_quota": 1073741824, <------------ stored in CCNG DB, not from Diego
      "disk_quota": 1073741824, <------------ stored in CCNG DB, not from Diego
      "fds_quota": 16384,
      "isolation_segment": null,
      "details": null,
      "instance_ports": [
        {
          "external": 61000,
          "internal": 8080,
          "external_tls_proxy_port": 61002,
          "internal_tls_proxy_port": 61001
        }
      ]
    },
```

This has unfortunate implications when the CLI displays app information:

```
name:              dora
requested state:   started
routes:            dora.vanilla-thorn.capi.land
last uploaded:     Mon 01 Feb 20:08:23 UTC 2021
stack:             cflinuxfs3
buildpacks:
        name             version   detect output   buildpack name
        ruby_buildpack   1.8.27    ruby            ruby

type:           web
sidecars:
instances:      1/1
memory usage:   1024M
     state     since                  cpu    memory        disk          details
#0   running   2021-02-01T20:08:48Z   0.5%   37.2M of 1G   91.9M of 1G
```

The combination of using *actual* used memory (in this case 37.2M) with desired state quota (here it is 1G) makes it seem like the 1G is how much memory is actually available.  We can even use this to create some nonsensical output: 

```
     state     since                  cpu    memory        disk          details
#0   running   2021-02-01T20:08:48Z   0.5%   37.2M of 1M   91.9M of 1M
```

In our docs, we don't describe if these fields are Actual or Desired state in our documentation:

![Screen Shot 2021-02-01 at 1 56 49 PM](https://user-images.githubusercontent.com/1445593/106523080-732a7880-6495-11eb-8dd0-fa5712e955f2.png)

Nevertheless, the whole endpoint seems to be oriented around Actual state and it is confusing to mix Desired state with Actual state.

We should change the instance stats reporter to only return diego information (where possible) rather than relying on the CCDB.

Relevant lines: https://github.com/cloudfoundry/cloud_controller_ng/blob/main/lib/cloud_controller/diego/reporters/instances_stats_reporter.rb#L44

We believe this is applicable to  `mem_quota`, `disk_quota`, `fds_quota` and potentially `uris`

@MerricdeLauney

**Comment by sethboyles** [comment]
This seems to have been introduced with the ZDT track changes

**Comment by Gerg** [comment]
@sethboyles @MerricdeLauney Correct me if I'm wrong, but wasn't this always an issue for disk, since updating disk didn't automatically trigger an update to the running LRP?

**Comment by Gerg** [comment]
I think it makes sense to swap these fields to show the actual values rather than desired, assuming we can get that information from Diego. If people want the desired scale information, they can get that from the process model directly.

It looks like we ported this behavior from v2:

v2 stats:
```json
{
  "0": {
    "state": "RUNNING",
    "isolation_segment": null,
    "stats": {
      "name": "dora",
      "uris": [
        "dora.leaf-crow.capi.land"
      ],
      "host": "10.244.0.138",
      "port": 61064,
      "uptime": 248,
      "mem_quota": 1048576,
      "disk_quota": 1048576,
      "fds_quota": 16384,
      "usage": {
        "time": "2021-02-02T19:19:54+00:00",
        "cpu": 0.004848659400713151,
        "mem": 36146744,
        "disk": 107900928
      }
    }
  }
}
```

v3 stats: 
```json
{
  "resources": [
    {
      "type": "web",
      "index": 0,
      "state": "RUNNING",
      "usage": {
        "time": "2021-02-02T19:18:42+00:00",
        "cpu": 0.004768601481830765,
        "mem": 36146744,
        "disk": 107900928
      },
      "host": "10.244.0.138",
      "uptime": 176,
      "mem_quota": 1048576,
      "disk_quota": 1048576,
      "fds_quota": 16384,
      "isolation_segment": null,
      "details": null,
      "instance_ports": [
        {
          "external": 61064,
          "internal": 8080,
          "external_tls_proxy_port": 61066,
          "internal_tls_proxy_port": 61001
        },
        {
          "external": 61065,
          "internal": 2222,
          "external_tls_proxy_port": 61067,
          "internal_tls_proxy_port": 61002
        }
      ]
    }
  ]
}
```

**Comment by Gerg** [comment]
It looks like the desired memory isn't available on the actual lrp:

```
cfdot actual-lrps -p f831615a-1ddd-4d29-aaf5-4a6b7dc38720-0dfce286-0c85-4c0f-831c-ecb0e0958349 | jq .
```
```json
{
  "process_guid": "f831615a-1ddd-4d29-aaf5-4a6b7dc38720-0dfce286-0c85-4c0f-831c-ecb0e0958349",
  "index": 0,
  "domain": "cf-apps",
  "instance_guid": "7ab4a809-b161-48b3-5059-148d",
  "cell_id": "cb6d55d2-d9c7-4e7b-b6e4-02645e81c875",
  "address": "10.244.0.138",
  "ports": [
    {
      "container_port": 8080,
      "host_port": 61064,
      "container_tls_proxy_port": 61001,
      "host_tls_proxy_port": 61066
    },
    {
      "container_port": 2222,
      "host_port": 61065,
      "container_tls_proxy_port": 61002,
      "host_tls_proxy_port": 61067
    }
  ],
  "instance_address": "10.255.234.41",
  "preferred_address": "HOST",
  "crash_count": 0,
  "state": "RUNNING",
  "since": 1612293345885014300,
  "modification_tag": {
    "epoch": "e6c7068b-762e-41e8-6142-e7af9fdcfe5a",
    "index": 2
  },
  "presence": "ORDINARY"
}
```

We could get it if we also fetched the desired lrp:
```
cfdot desired-lrps | grep "f831615a-1ddd-4d29-aaf5-4a6b7dc38720-0dfce286-0c85-4c0f-831c-ecb0e0958349" | jq .
```
```json
{
  "process_guid": "f831615a-1ddd-4d29-aaf5-4a6b7dc38720-0dfce286-0c85-4c0f-831c-ecb0e0958349",
  "domain": "cf-apps",
  "rootfs": "preloaded:cflinuxfs3",
  "instances": 1,
  "env": ["..."],
  "setup": {"...": "..."},
  "action": {"...": "..."},
  "start_timeout_ms": 60000,
  "disk_mb": 256,
  "memory_mb": 256,
  "cpu_weight": 3,
  "privileged": false,
  "ports": [8080, 2222],
  "routes": {"...": "..."},
  "log_source": "CELL",
  "log_guid": "c163959f-6a6d-428c-9c4c-4ef892f644a8",
  "metrics_guid": "c163959f-6a6d-428c-9c4c-4ef892f644a8",
  "annotation": "1612293500.0",
  "egress_rules": ["..."],
  "modification_tag": {"...": "..."},
  "cached_dependencies": ["..."],
  "legacy_download_user": "vcap",
  "trusted_system_certificates_path": "/etc/cf-system-certificates",
  "network": {"...": "..."},
  "max_pids": 1024,
  "certificate_properties": {"...": "..."},
  "metric_tags": {"...": "..."}
}
```

**Comment by Gerg** [comment]
On the metrics side, the trafficcontroller interface does have `memoryBytesQuota` and `diskBytesQuota`: https://github.com/cloudfoundry/cloud_controller_ng/blob/60a2a9e30dc6f743fb107592e898e83d6f0ee0c1/lib/traffic_controller/models/metric.pb.rb#L43-L44

However, we don't appear to set them when converting from Log Cache (like we do for `memoryBytes` and `diskBytes`): 
https://github.com/cloudfoundry/cloud_controller_ng/blob/a2f5298a9ab0f4cc6116848659956b0c4e951656/lib/logcache/traffic_controller_decorator.rb#L80-L97

It appears that we are getting that metric from Log Cache, but we are dropping it:
```rb
Logcache::TrafficControllerDecorator.new(CloudController::DependencyLocator.instance.logcache_client).send(:get_container_metrics, {start_time: TimeUtils.to_nanoseconds(Time.now - 2.minutes), end_time: TimeUtils.to_nanoseconds(Time.now), source_guid: "c163959f-6a6d-428c-9c4c-4ef892f644a8"}).last.to_json
```
```json
{
  "timestamp": "1612298249938174530",
  "sourceId": "c163959f-6a6d-428c-9c4c-4ef892f644a8",
  "deprecatedTags": {},
  "gauge": {
    "metrics": {
      "disk_quota": {
        "unit": "bytes",
        "value": 268435456
      },
      "cpu": {
        "unit": "percentage",
        "value": 0.45236124942758776
      },
      "memory_quota": {
        "unit": "bytes",
        "value": 268435456
      },
      "disk": {
        "unit": "bytes",
        "value": 107900928
      },
      "memory": {
        "unit": "bytes",
        "value": 36154026
      }
    }
  },
  "instanceId": "0",
  "tags": {
    "index": "cb6d55d2-d9c7-4e7b-b6e4-02645e81c875",
    "app_id": "c163959f-6a6d-428c-9c4c-4ef892f644a8",
    "process_instance_id": "7ab4a809-b161-48b3-5059-148d",
    "source_id": "c163959f-6a6d-428c-9c4c-4ef892f644a8",
    "ip": "10.244.0.138",
    "app_name": "dora",
    "job": "diego-cell",
    "instance_id": "0",
    "deployment": "cf",
    "space_name": "space",
    "process_type": "web",
    "organization_name": "org",
    "origin": "rep",
    "space_id": "d0f2ecd5-7d6d-4855-abd5-e0a59d5194bb",
    "process_id": "f831615a-1ddd-4d29-aaf5-4a6b7dc38720",
    "organization_id": "6684c692-8311-4890-896d-f70aa207c633"
  }
}
```

**Comment by Gerg** [comment]
Based on the above, the easiest course of action is probably to use the `memory_quota` and `disk_quota` metrics from log cache. Note that this might not work cleanly for deployments still using traffic controller instead of log cache (not sure if that is still supported anymore) and for cf-for-k8s.

cc @tcdowney @jspawar

**Comment by JenGoldstrich** [comment]
I think this issue was resolved in [CAPI Release 1.108.0](https://github.com/cloudfoundry/capi-release/releases/tag/1.108.0), updating release notes to include this issue and closing this!
